src/app/operators/operators.component.html

```diff
@@ -5,12 +5,18 @@
                [fixedInViewport]="true"
                class="operator-list-sidenav"
                #operatorSidenav>
+    <form>
+      <mat-form-field class="full-width-form">
+        <mat-label>Search</mat-label>
+        <input matInput value="" [formControl]="searchInput">
+        <mat-icon matSuffix>search</mat-icon>
+      </mat-form-field>
+    </form>
     <mat-nav-list *ngFor="let category of categories"
                   class="operator-list">
-      <h3 mat-subheader
-          class="category-subheader">{{ category }}</h3>
+      <h3 mat-subheader class="category-subheader" *ngIf="(groupedOperators[category] | filter: searchInput.value).length > 0">{{ category }}</h3>
```

Comment:
This wont have a good performance as the # of operators increase, as the filtering is run 2 times

Reply:
yeah I had that idea too, but last time I was simply to supid to implement it that way. But you are definetly right, I will give it another try :D